airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.kt

```diff
@@ -9,36 +9,53 @@ import io.airbyte.cdk.db.jdbc.JdbcDatabase
 import io.airbyte.cdk.integrations.destination.NamingConventionTransformer
 import io.airbyte.cdk.integrations.destination.jdbc.ColumnDefinition
 import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition
-import io.airbyte.cdk.integrations.destination.jdbc.typing_deduping.JdbcDestinationHandler.Companion.fromIsNullableIsoString
 import io.airbyte.integrations.base.destination.typing_deduping.BaseDestinationV1V2Migrator
 import io.airbyte.integrations.base.destination.typing_deduping.CollectionUtils.containsAllIgnoreCase
 import io.airbyte.integrations.base.destination.typing_deduping.NamespacedTableName
 import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.destination.snowflake.SnowflakeDatabaseUtils.fromIsNullableSnowflakeString
+import java.sql.SQLException
 import java.util.*
 import lombok.SneakyThrows
+import org.json.JSONObject
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
 @SuppressFBWarnings("NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE")
 class SnowflakeV1V2Migrator(
     private val namingConventionTransformer: NamingConventionTransformer,
     private val database: JdbcDatabase,
     private val databaseName: String
 ) : BaseDestinationV1V2Migrator<TableDefinition>() {
+
+    private val LOGGER: Logger =
+        LoggerFactory.getLogger(SnowflakeV1V2Migrator::class.java)
+
     @SneakyThrows
     @Throws(Exception::class)
     override fun doesAirbyteInternalNamespaceExist(streamConfig: StreamConfig?): Boolean {
-        return database
-            .queryJsons(
+
+        var showSchemaResult : List<JsonNode> = listOf()
+
+        try {
+            val showSchemaQuery = String.format(
                 """
-                SELECT SCHEMA_NAME
-                FROM information_schema.schemata
-                WHERE schema_name = ?
-                AND catalog_name = ?;
-                
+               SHOW SCHEMAS LIKE '%s' IN DATABASE "%s";
                 """.trimIndent(),
                 streamConfig!!.id.rawNamespace,
-                databaseName
+                databaseName,
             )
-            .isNotEmpty()
+
+            showSchemaResult = database.queryJsons(
```

Comment:
val here. There's no need for a var. I'd expect there's no need to close the stream either